File: ROS/catkin_ws/src/my_robot_tutorials/src/add_two_ints_client.cpp
#include <ros/ros.h>
#include <rospy_tutorials/AddTwoInts.h>

int main(int argc, char **argv) {

	ros::init(argc, argv, "add_two_ints_client");
	ros::NodeHandle nh;

	// create a ServiceClient object.
	// Arguments: service name
	ros::ServiceClient client = nh.serviceClient<rospy_tutorials::AddTwoInts>("/add_two_ints");

	rospy_tutorials::AddTwoInts srv;
	srv.request.a = 12;
	srv.request.b = 5;

	// if will call the service
	if (client.call(srv)) {
		// proccess the data
		ROS_INFO("Returned sum is %d", (int)srv.response.sum);
	}
	else {
		// proccess failed
		ROS_WARN("Service call failed");
	}
}

File: ROS/catkin_ws/src/my_robot_tutorials/src/scripts/add_two_ints_client.py
#!/usr/bin/env python3

import rospy
from rospy_tutorials.srv import AddTwoInts

if __name__ == '__main__':
	rospy.init_node("add_two_ints_client")

	# we wait for service to start before calling for it; otherwise this will throw an error if service is not initialized
	rospy.wait_for_service("/add_two_ints")

	# creating client
	# if the client fails to call the client it will throw an exception.
	try:
		# creating a client using ServiceProxy with arguments:
		# name of topic, topic type
		add_two_ints = rospy.ServiceProxy("/add_two_ints", AddTwoInts)
		# calling the client and capturing the output
		response = add_two_ints(2,6)
		rospy.loginfo("Sum is: " + str(response.sum))

	except rospy.ServiceException as e:
		# catching the exception
		rospy.logwarn("Service failed: " + str(e))